### PR TITLE
Adding workflow for pushing container to github packages

### PR DESCRIPTION
**Pull request by FredDodd6**
Adding GitHub Action workflow to push images to GitHub Packages.
change tag namespace to include ghcr.io

### CODE REVIEW

File: docker/Makefile [hunk 1]
```diff
@@ -23,6 +23,7 @@ ub-12.04-base.$(TAG): base/ubuntu/12.04/Dockerfile
 	--tag $(TAG_NAMESPACE)/ub-12.04-base:$(TAG) --file $< ./base
 	touch $@
 
+
```
Comment (kjsanger) [inline docker/Makefile:26]: Please remove the extra whitespace changes.

**Review by kjsanger — CHANGES_REQUESTED** [review]

**Comment by kjsanger** [comment]
Is this still in play, @FredDodd6 , @dkj ?

**Comment by dkj** [comment]
> Is this still in play, @FredDodd6 , @dkj ?

I think it gives us an extra path with one less component (DockerHub) - so only a good thing? The last problem I was aware of was flaky connections to pick up official iRODS pacakges...

**Comment by kjsanger** [comment]
I'll leave open. I was asking because ~1 month since last activity.

File: docker/Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-TAG=$(shell git describe)
+TAG=$(shell git describe --always)
 LABEL_NAMESPACE=uk.ac.sanger
-TAG_NAMESPACE=wsinpg
+TAG_NAMESPACE=ghcr.io/$(REPO_OWNER)
```
Comment (dkj) [inline docker/Makefile:3]: How about 
```suggestion
ifdef REPO_OWNER
TAG_NAMESPACE?=ghcr.io/$(REPO_OWNER)
else
TAG_NAMESPACE?=wsinpg
endif
```
allow us to run with a `TAG_NAMESPACE` suitable for DockerHub, or of choice with `make TAG_NAMESPACE=foo all`

File: .github/workflows/pushing_container_image.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            make REPO_OWNER=${{ env.REPO_OWNER }}
+     
+      - name: push images
+        run: |      
+            docker images
+            export TAG=$(git describe --always)
+            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-12.04-base:$TAG
+            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-16.04-base:$TAG
+            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-18.04-base:$TAG
+            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-12.04-conda:$TAG
```
Comment (dkj) [inline .github/workflows/pushing_container_image.yml:47]:
```suggestion
            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-16.04-base:$TAG
            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-18.04-base:$TAG
            docker push ghcr.io/${{ env.REPO_OWNER }}/ub-16.04-conda:$TAG
```
Update following removal of Precise (12.04) #15

**Pull request merged**